dbcrossbarlib/src/drivers/shopify/mod.rs

```diff
@@ -0,0 +1,96 @@
+//! Driver for working with Shopify REST endpoints.
+//!
+//! We include this because it forced us to face a number of interesting issues, including struct types,
+
+use std::{fmt, str::FromStr};
+
+use crate::common::*;
+
+mod json_to_csv;
+mod local_data;
+
+use local_data::local_data_helper;
+
+/// A Shopify REST endpoint.
+#[derive(Clone, Debug)]
+pub(crate) struct ShopifyLocator {
+    url: Url,
+}
+
+impl ShopifyLocator {
+    /// Convert this locator to a `https` URL.
+    fn to_https_url(&self) -> Result<Url> {
+        assert!(self.url.as_str().starts_with(Self::scheme()));
+        let https_str =
+            format!("https:{}", &self.url.as_str()[Self::scheme().len()..]);
+        let https_url = https_str
+            .parse::<Url>()
+            .with_context(|_| format_err!("could not set URL scheme for {}", self))?;
+        Ok(https_url)
+    }
+}
+
+#[test]
+fn to_https_url() {
+    let loc = ShopifyLocator::from_str(
+        "shopify://example.myshopify.com/admin/api/2020-04/orders.json",
```

Comment:
ah FQDN not just shop name

Reply:
Shop names are always given as FQDNs by the Shopify API, and I don't dare to tamper with that.